wgpu-hal/src/gles/adapter.rs

```diff
@@ -1232,7 +1232,10 @@ impl crate::Adapter for super::Adapter {
                 } else {
                     vec![wgt::PresentMode::Fifo] //TODO
                 },
-                composite_alpha_modes: vec![wgt::CompositeAlphaMode::Opaque], //TODO
+                composite_alpha_modes: vec![
+                    wgt::CompositeAlphaMode::Opaque,
+                    wgt::CompositeAlphaMode::PreMultiplied,
+                ], //TODO
```

Comment:
I don't think this is right - without telling this to the runtime at all we can't know if these modes are actually supported. I'm not sure how this works with the various platforms, but I don't think we can just change it.

Reply:
It look like `premultipliedAlpha` is set to true by default https://registry.khronos.org/webgl/specs/latest/1.0/#5.2 and we do not seem to change it

Reply:
It looks like supporting multiple modes could be tricky if we want it to be handled at the WebGL context level, because we'd need to create the context (wgpu's `Surface`) with opaque vs. premultiplied. There seem like a couple options
- If we exposed a way to specify whether alpha and premultiplied should be used when creating the initial surface/context, then we could change `composite_alpha_modes` to return `Opaque`/`PreMultiplied` depending on whatever the surface is currently using.
- If we always use premultiplied on the context, I guess we could at least handle both opaque and premultiplied for srgb presentation by handling it in `srgb_present.frag`. This wouldn't work for non-srgb though where we blit the framebuffer (unless we went through a shader there for the opaque case).